pandas-stubs/core/tools/numeric.pyi

```diff
@@ -1 +1,64 @@
-def to_numeric(arg, errors: str = ..., downcast=...): ...
+from typing import (
+    Literal,
+    overload,
+)
+
+import numpy as np
+import pandas as pd
+
+from pandas._typing import (
+    IgnoreRaiseCoerce,
+    Scalar,
+    npt,
+)
+
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> int | float: ...
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["ignore"],
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> Scalar: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: IgnoreRaiseCoerce = ...,
+    *,
+    downcast: Literal["integer", "signed", "unsigned", "float"],
+) -> npt.NDArray: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: Literal["ignore"],
+    downcast: None = ...,
+) -> npt.NDArray: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: None = ...,
+) -> npt.NDArray[np.intp] | npt.NDArray[np.float_]: ...
+@overload
+def to_numeric(
+    arg: pd.Series,
+    errors: IgnoreRaiseCoerce = ...,
+    *,
+    downcast: Literal["integer", "signed", "unsigned", "float"],
+) -> pd.Series: ...
+@overload
+def to_numeric(
+    arg: pd.Series,
+    errors: Literal["ignore"],
+    downcast: None = ...,
+) -> pd.Series: ...
```

Comment:
again, it seems to me that these 2 overloads don't need to be differentiated, because they return the same result, independent of the arguments, but maybe I'm missing something.

Reply:
Same issue here.  There are certain combinations where the dtype can be object or something numeric, and possibly something else (hard to know).  For standard cases, you either get int or float.

Reply:
Same response as above.  Could just do:
```python
def to_numeric(
    arg: pd.Series,
    errors: IgnoreRaiseCoerce = ...,
    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...
) -> pd.Series: ...
```

Reply:
Removeing the union returns let me simplify the other cases.